apps/client/src/hoc/withChecker.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useContext, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+import { TopicOutputType } from "../../sttr_build/types";
+import { Topic, topics } from "../config";
+import { autofillMap, autofillResolvers } from "../config/autofill";
+import { CheckerContext, SessionContext, SessionDataType } from "../context";
+import ErrorPage from "../pages/ErrorPage";
+import LoadingPage from "../pages/LoadingPage";
+import RedirectPage from "../pages/RedirectPage";
+import getChecker from "../sttr_client";
+
+export default (Component: any) => () => {
+  const sessionContext = useContext(SessionContext) as SessionDataType;
+  const checkerContext = useContext(CheckerContext);
+  const [checker, setChecker] = useState(checkerContext.checker);
+  const [error, setError] = useState();
+  const { slug } = useParams();
+
+  const topic = topics.find((t) => t.slug === slug);
+
+  useEffect(() => {
+    // Default settings to be able to open the CheckerPage
+    if (!sessionContext[slug]) {
+      sessionContext.setSessionData([
+        slug,
+        { activeComponents: ["locationInput"], finishedComponents: [] },
+      ]);
+    }
+  });
+  useEffect(() => {
+    initChecker();
+  });
+
+  const initChecker = async () => {
```

Comment:
Nice that you converted to tsx and in one week this whole file will be deleted 🤣

Reply:
useChecker will be largely the same 😜